static/app/views/explore/hooks/useGroupBys.tsx

```diff
@@ -12,37 +14,39 @@ import {useSpanTags} from '../contexts/spanTagsContext';
 interface Options {
   location: Location;
   navigate: ReturnType<typeof useNavigate>;
-  tags: TagCollection;
+  tagsResults: UseApiQueryResult<TagCollection, RequestError>;
 }
 
-export function useGroupBys(): [Field[], (fields: Field[]) => void] {
+export function useGroupBys(): [Field[], (fields: Field[]) => void, boolean] {
   const location = useLocation();
   const navigate = useNavigate();
-  const tags = useSpanTags();
-  const options = {location, navigate, tags};
+  const tagsResults = useSpanTags();
+  const options = {location, navigate, tagsResults};
 
   return useGroupBysImpl(options);
 }
 
 function useGroupBysImpl({
   location,
   navigate,
-  tags,
-}: Options): [Field[], (fields: Field[]) => void] {
+  tagsResults,
+}: Options): [Field[], (fields: Field[]) => void, boolean] {
+  const {data: tags, isLoading: isLoadingTags} = tagsResults;
+
   const groupBys = useMemo(() => {
     const rawGroupBys = decodeList(location.query.groupBy);
 
     // Filter out groupBys that are not in span field supported tags
-    const validGroupBys = rawGroupBys.filter(
-      groupBy => groupBy === '' || tags.hasOwnProperty(groupBy)
-    );
+    const validGroupBys = isLoadingTags
+      ? rawGroupBys
+      : rawGroupBys.filter(groupBy => groupBy === '' || tags?.hasOwnProperty(groupBy));
```

Comment:
If `isLoadingTags` is true, I feel like it make might more sense to have `validGroupBys == []` or `validGroupBys == ['']`, since we don't know if any of the values in `rawGroupBys` are valid yet. 

Either way, since we're now propagating the `isLoading` state, anyone using this hook should be checking `isLoading` before using the `groupBys` result.